Meteor 1.6

jshimko opened the pull request:
- update to Meteor 1.6
- update all npm dependencies (except React/ReactDOM)
- remove unused (?) `BlazeLayout` component to fix build/compile issue
- update Docker base to v3.0.0 (which [added Meteor 1.6 support](https://github.com/reactioncommerce/base/commit/db87343b2cafac752dc7512189e3d71b36022606))

Outstanding issues:
- evaluate whether we need to support `BlazeLayout` anymore.  cc: @mikemurray 

Other than the one remaining discussion above, builds work and tests pass.  We should be pretty close here (if not already good to go).

To test this PR:

```sh
reaction init -b meteor-1.6 reaction-1.6
cd reaction-1.6
reaction
```

mikemurray commented:
@jshimko looking at the use`BlazeLayout` now. Will report back soon.

mikemurray commented:
@jshimko I couldn't find anywhere `BlazeLayout` was used. I'd say remove it, and if the app still functions as expected without it, then we should be good.

jshimko commented:
Ok.  It has already been removed as part of this PR.  Thanks @mikemurray.

So I guess the last remaining thing to confirm is that all of the updated npm packages aren't causing any problems.  Several of them were major version upgrades.  I tested a bunch, but it would be helpful if a few other people could do the same so we don't introduce any breaking changes.

spencern commented:
@jshimko I'll give this a thorough run through today. 

Should we mute bithound on React and React DOM if we're not updating them yet?

hrath2015 commented:
One good news is for windows platform(10) users.... Reaction started without the   **FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory** issue.  👍

machikoyasuda commented:
Related to @zenweasel's https://github.com/reactioncommerce/reaction/issues/3235 issue and https://github.com/reactioncommerce/reaction/issues/3207

machikoyasuda commented:
also closes https://github.com/reactioncommerce/reaction/issues/3029

fisshy commented:
This one works on Windows 10 x64  - using Meteor 1.6.

zenweasel commented:
Will give this a thorough testing today

zenweasel commented:
I get this error on startup (this is a fresh clone into a separate directory):

```
Error: The module '/Users/brenthoover/Projects/js/reaction16/node_modules/dtrace-provider/build/Release/DTraceProviderBindings.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 46. This version of Node.js requires
NODE_MODULE_VERSION 57. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at Object.Module._extensions..node (module.js:641:18)
    at Module.load (module.js:531:32)
    at tryModuleLoad (module.js:494:12)
    at Function.Module._load (module.js:486:3)
    at Module.require (module.js:556:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/brenthoover/Projects/js/reaction16/node_modules/dtrace-provider/dtrace-provider.js:18:23)
    at Module._compile (module.js:612:30)
    at Object.Module._extensions..js (module.js:623:10)
    at Module.load (module.js:531:32)
```

zenweasel commented:
Also this during app startup
```
(node:22148) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): MongoError: ns not found
(node:22148) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

zenweasel commented:
These errors in the browser console. App never starts:
```
Uncaught ReferenceError: Buffer is not defined
    at util.js (modules.js:60033)
    at fileEvaluate (modules-runtime.js:343)
    at require (modules-runtime.js:238)
    at _stream_readable.js (modules.js:54532)
    at fileEvaluate (modules-runtime.js:343)
    at require (modules-runtime.js:238)
    at readable.js (modules.js:54447)
    at fileEvaluate (modules-runtime.js:343)
    at require (modules-runtime.js:238)
    at index.js (modules.js:54325)
app.js:114707 Uncaught TypeError: Cannot read property 'marketplaceEnabled' of undefined
    at app.js:114707
    at Tracker.Computation._compute (tracker.js:339)
    at new Tracker.Computation (tracker.js:229)
    at Object.Tracker.autorun (tracker.js:613)
    at app.js:114704
    at maybeReady (meteor.js:821)
```

zenweasel commented:
Upgrading in place seems to work fine

aaronjudd commented:
Fresh clone "working" for me, but I am seeing this message  as well: 
```
(node:25986) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): MongoError: ns not found
```

jshimko commented:
Ok, so the "unhandled promise rejection" is because some upstream code (the Mongo driver) is executing a Promise without using a `catch` statement.  Newer versions of Node no longer swallow that poor form.  Fortunately, there's an easy way to make that error a bit more useful.  See [my last commit](https://github.com/reactioncommerce/reaction/pull/3243/commits/cd1322c77c5f77fe998c3924d4bc8e18296554d0).

Right at the top of `/server/main.js`...

```js
import { Logger } from "/server/api";

process.on("unhandledRejection", (err) => {
  Logger.error(err);
});
```
More info: https://nodejs.org/docs/latest-v8.x/api/process.html#process_event_unhandledrejection

So this doesn't exactly solve the problem, but it makes it a lot more clear where it's coming from by logging out the stack trace.

```
15:17:54.583Z ERROR Reaction: ns not found
  MongoError: ns not found
      at Function.MongoError.create (/Users/Jeremy/.meteor/packages/npm-mongo/.2.2.33.4z2l93.ajzfg++os+web.browser+web.cordova/npm/node_modules/mongodb-core/lib/error.js:31:11)
      at /Users/Jeremy/.meteor/packages/npm-mongo/.2.2.33.4z2l93.ajzfg++os+web.browser+web.cordova/npm/node_modules/mongodb-core/lib/connection/pool.js:497:72
      at authenticateStragglers (/Users/Jeremy/.meteor/packages/npm-mongo/.2.2.33.4z2l93.ajzfg++os+web.browser+web.cordova/npm/node_modules/mongodb-core/lib/connection/pool.js:443:16)
      at Connection.messageHandler (/Users/Jeremy/.meteor/packages/npm-mongo/.2.2.33.4z2l93.ajzfg++os+web.browser+web.cordova/npm/node_modules/mongodb-core/lib/connection/pool.js:477:5)
      at Socket.<anonymous> (/Users/Jeremy/.meteor/packages/npm-mongo/.2.2.33.4z2l93.ajzfg++os+web.browser+web.cordova/npm/node_modules/mongodb-core/lib/connection/connection.js:331:22)
      at emitOne (events.js:115:13)
      at Socket.emit (events.js:210:7)
      at addChunk (_stream_readable.js:263:12)
      at readableAddChunk (_stream_readable.js:250:11)
      at Socket.Readable.push (_stream_readable.js:208:10)
      at TCP.onread (net.js:597:20)
```

That's obviously an issue in [the Mongo driver that Meteor uses](https://github.com/mongodb-js/mongodb-core), but it's unclear if it's actually breaking anything and it only happens on the initial build.  Subsequent builds don't cause that (for me).

As for all of the other errors reported by @spencern and @zenweasel, they sound like you may have accidentally installed node_modules on the wrong branch (which means a totally different version of Node).  @zenweasel, I know you said that was a fresh clone, but the only time I've ever seen that error is when you switch Node versions and try to use existing node_module binaries from a previous Node version's build.  For example, @spencern if you ran the commands just like you mentioned above, you definitely installed node_modules with the wrong Node version.

You ran...

```sh
# clones master and runs `meteor npm install` (currently using Node 4.8.4)
reaction init meteor-1.6   

# switch to Meteor 1.6 and Node 8, but does NOT do the required re-install of node_modules
git checkout meteor-1.6  
reaction pull meteor-1.6

# builds the app with the wrong node_modules, almost guaranteed issues
reaction
```

FYI, @spencern you can do the exact same steps in one command.

```sh
reaction init -b <branch> <outputDir>

# so, to clone my PR branch and install node_modules with Meteor 1.6 and Node 8.8.1
reaction init -b meteor-1.6 reaction-1.6
```

Anyway, I just did a fresh clone with the command above and the only issue I'm seeing is that Mongo error (which now has a properly formatted stack trace).  So we probably have to file an issue with Meteor on that one.  As for the other stuff, I'm still unable to reproduce any of it.

jshimko commented:
Just to make sure we're all doing these steps in the right order, lets isolate every command that `reaction-cli` is running for you under the hood.  For anyone experiencing the above issues (other than the Mongo issue), please do the steps below to confirm you build binaries with the correct Node version.

```sh
git clone -b meteor-1.6 --single-branch https://github.com/reactioncommerce/reaction.git one-six

cd one-six

meteor npm install

reaction
```

@hrath2015 your global Node version shouldn't matter because Meteor uses whatever version it has bundled with it.

spencern commented:
Following @jshimko's steps, the client loaded properly the first time and the only error I saw was the above-noted: `MongoError: ns not found`

I did see that error twice, is it possible that we're causing it within migrations by attempting action on a collection that doesn't exist?

I'll run 1.6 through in more detail now, but it does appear my earlier issues were due to incompatible `node_modules`

jshimko commented:
Good point.  I hadn't considered that it might be because of migrations.  Definitely worth investigating.

I did check Meteor issues/PR's and didn't find any mention of the issue from other users, so it is likely something to do with how we're using the Mongo API.

hrath2015 commented:
@jshimko  Thanks it works after following  the steps. Able to load client.  Will do more testing and report if find anything. 👍

zenweasel commented:
We need to make sure we update docs/tutorials to reflect the removal of BlazeLayout because we reference that a lot. We don't really use it in core code but it is how we tell people to change layouts I believe.

spencern commented:
Diving in a little deeper, the `MongoError: ns not found` is caused by one of our event hooks running during `afterCoreInit`

Commenting out the `Hooks.Events.run("afterCoreInit"); eliminates the issue. 

https://github.com/reactioncommerce/reaction/blob/meteor-1.6/server/api/core/core.js#L57

I believe that there may be two separate `ns not found` issues, one in the migrations (which run afterCoreInit), and one in another hook running afterCoreInit.

jshimko commented:
After several hours of further testing of that Mongo error, it appears it's from any time something runs `Collection.remove({})` on a collection that doesn't exist yet.  And it only happens on first startup.  Any time after that, all collections already exist.  @spencern, it was not because the collection variables were undefined, so I reverted your last commit.

What it comes down to is we would have to go through the entire app and find instances that _may_ possibly do a remove operation too quickly on first startup and we'd have to guard against it like this (or something like it)...

```js
  if (MyCollection.find().count()) {
    MyCollection.remove({});
  }
```

That's obviously a little gross.  I think we should be able to safely swallow that error since having no collection yet is pretty much identical to removing all documents from a that collection.  So instead I opted to debug log it and otherwise ignore it...

```js
// handle any unhandled Promise rejections
process.on("unhandledRejection", (err) => {
  // If we get an error from the Mongo driver because something tried to drop all
  // documents in a collection before it existed, log it out as debug info.
  // Otherwise, log whatever happened as an error.
  if (err.name === "MongoError" && err.message === "ns not found") {
    Logger.debug("Attempted to remove document from Mongo collection that doesn't exist yet");
    Logger.debug(err);
  } else {
    Logger.error(err);
  }
});
```

All errors gone.  Everything appears to work as expected.  Please give the whole app another test to confirm.

aaronjudd commented:
Don't we also want to / need to commit all `package-lock.json` as well, as this is part will get recreated by every install otherwise. 

```
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN reaction-connectors-shopify@1.0.0 No repository field.
```

jshimko commented:
The most up to date package-lock.json is there.

https://github.com/reactioncommerce/reaction/blob/0e704b7a3702782267d75c101e8e67f34f24cbff/package-lock.json

jshimko commented:
Index update failures due to nonexistent collections on initial start are now explicitly handled.

https://github.com/reactioncommerce/reaction/pull/3243/files#diff-ac1b95bb2084a909c414d91a9106a66c

I think we might be good to go here.

zenweasel commented on imports/plugins/included/search-mongo/server/methods/searchcollections.js:
```diff
@@ -54,6 +54,25 @@ function getSearchLanguage() {
+ * @return {undefined}   doesn't return anything
+ */
+function handleRawCollectionFailures(error) {
+  // If we get an error from the Mongo driver because something tried to drop al
```
typo in comment

jshimko replied:
Yeah, I fixed that 3 days ago.  https://github.com/reactioncommerce/reaction/pull/3243/commits/881f2dc3243d1143a637f789f8178da7b46e5420

Not sure how that was still showing here.

hrath2015 commented:
On windows `babel-plugin-lodash@3.3.1`  is not working.  Getting all sorts of errors. Reverted back to `@3.2.11` worked.   

```
While processing files with ecmascript (for target web.browser):

   C:\Users\xxx\AppData\Local\.meteor\packages\ecmascript\0.9.0\plugin.compile-ecmascript.os\npm\node_modules\meteor\babel-compiler\node_modules\babel-core\lib\transformation\plugin.js:131:13:
   Plugin 0 specified in "foreign" provided an invalid property of
   "__wrapped__"
   at Plugin.init
   (C:\Users\xxx\AppData\Local\.meteor\packages\ecmascript\0.9.0\plugin.compile-ecmascript.os\npm\node_modules\meteor\babel-compiler\node_modules\babel-core\lib\transformation\plugin.js:131:13)
   at Function.normalisePlugin
   (C:\Users\xxx\AppData\Local\.meteor\packages\ecmascript\0.9.0\plugin.compile-ecmascript.os\npm\node_modules\meteor\babel-compiler\node_modules\babel-core\lib\transformation\file\options\option-manager.js:152:12)
   at
   C:\Users\xxx\AppData\Local\.meteor\packages\ecmascript\0.9.0\plugin.compile-ecmascript.os\npm\node_modules\meteor\babel-compiler\node_modules\babel-core\lib\transformation\file\options\option-manager.js:184:30
```

jshimko commented:
@hrath2015 `babel-plugin-lodash@3.3.2` addresses that issue.  Updated and [all set on CircleCI](https://circleci.com/gh/reactioncommerce/reaction/9774)

zenweasel commented:
When trying to upgrade in place I get this error:
```
{ Error: Cannot find module './build/Release/DTraceProviderBindings'
    at Function.Module._resolveFilename (module.js:513:15)
    at Function.Module._load (module.js:463:25)
    at Module.require (module.js:556:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/brenthoover/Projects/js/reaction/node_modules/bunyan/node_modules/dtrace-provider/dtrace-provider.js:18:23)
    at Module._compile (module.js:612:30)
    at Object.Module._extensions..js (module.js:623:10)
    at Module.load (module.js:531:32)
    at tryModuleLoad (module.js:494:12)
    at Function.Module._load (module.js:486:3) code: 'MODULE_NOT_FOUND' }
```

but app starts....

zenweasel commented:
Placing an order with Stripe/Flat Rate/Avalara and get this order on the server console, but I **do not** get it every time.
```
(node:80506) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 3): TypeError: Cannot read property 'userId' of undefined
(node:80506) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit cod
```

zenweasel commented:
One lint error:
```
/Users/brenthoover/Projects/js/reaction/imports/plugins/core/router/lib/router.js
  541:19  error  Empty block statement  no-empty
```

jshimko commented:
@zenweasel 

**RE:...**
```
{ Error: Cannot find module './build/Release/DTraceProviderBindings'
    at Function.Module._resolveFilename (module.js:513:15)
    at Function.Module._load (module.js:463:25)
    at Module.require (module.js:556:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/brenthoover/Projects/js/reaction/node_modules/bunyan/node_modules/dtrace-provider/dtrace-provider.js:18:23)
    at Module._compile (module.js:612:30)
    at Object.Module._extensions..js (module.js:623:10)
    at Module.load (module.js:531:32)
    at tryModuleLoad (module.js:494:12)
    at Function.Module._load (module.js:486:3) code: 'MODULE_NOT_FOUND' }
```
That means your node_modules didn't properly install.  That module in particular needs to be compiled with each different major version of Node, so I suspect that has something to do with it.  Try a full reset with `reaction reset -y`.

**RE: ...**

```
(node:80506) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 3): TypeError: Cannot read property 'userId' of undefined
(node:80506) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit cod
```

I've [added back exception handling for unhandled Promise rejections](https://github.com/reactioncommerce/reaction/pull/3243/commits/79fc355a9b11ae7958156e7f7fda0c5d90e7c703), so you should now see a proper stack trace when that happens.  The Stripe lib is all Promise based, so I pretty much guarantee we're using one of their methods without adding proper error handling via a `catch`.

**RE:  empty block lint error**

Fixed. 28a53fb9f793ce7079a9e8e3b246c9acf167d4a3

hrath2015 commented:
getting this error on Order page.  
```
modules.js?hash=f100118521cff1dfeeff0f4c45f595bfaba6a3b8:35029 Warning: Failed prop type: Invalid prop `noDataText` supplied to `ReactTable`, expected a ReactNode.
    in ReactTable (created by Reaction(SortableTable))
    in Reaction(SortableTable) (created by Reaction(EmailLogs))
    in div (created by Reaction(CardBody))
    in VelocityTransitionGroupChild (created by TransitionGroup)
    in span (created by TransitionGroup)
    in TransitionGroup (created by VelocityTransitionGroup)
    in VelocityTransitionGroup (created by Reaction(CardBody))
    in Reaction(CardBody) (created by Reaction(EmailLogs))
    in div (created by Reaction(Card))
    in Reaction(Card) (created by Reaction(EmailLogs))
    in div (created by Reaction(CardGroup))
    in Reaction(CardGroup) (created by Reaction(EmailLogs))
    in Reaction(EmailLogs) (created by withProps(EmailLogs))
    in withProps(EmailLogs) (created by Tracker(withProps(EmailLogs)))
    in Tracker(withProps(EmailLogs))
```

janus-reith commented:
Just pulled the latest commits from meteor-1.6 branch, which worked for me before.

Now I'm getting `Uncaught TypeError: Cannot read property 'enter' of undefined`, thrown by `Reaction.Router.triggers.enter([notifySegment, notifyGoogleAnalytics, notifyMixpanel]);` in /plugins/included/analytics/client/startup.js

I deleted anayltics, and now noticed there're several other issues starting with:
`_Reaction.Router.getParam is not a function`

janus-reith commented:
I Just pulled the latest commits from meteor-1.6 branch - Which worked before.
Now I'm getting `Uncaught TypeError: Cannot read property 'enter' of undefined`, thrown by `Reaction.Router.triggers.enter([notifySegment, notifyGoogleAnalytics, notifyMixpanel]);` in /plugins/included/analytics/client/startup.js

I deleted anayltics, and now noticed there re several other issues starting with:
`_Reaction.Router.getParam is not a function`

jshimko commented:
@hrath2015 known issue https://github.com/reactioncommerce/reaction/issues/3219

@janus-reith is it possible node_modules haven't been properly installed?  Try `reaction reset -y` while on this branch.

janus-reith commented:
is reaction reset neccesary?
I manually deleted node_modules and npm install, that didn't solve it.
Then I tried deleting package-lock.json and reinstalling, that also didn't help.

janus-reith commented:
I just did a fresh checkout into a new folder with
`reaction init -b meteor-1.6 reaction`

and I get the same `Uncaught TypeError: Cannot read property 'enter' of undefined`

hrath2015 commented:
also getting similar issue... tried reset and new fresh install
```
Uncaught TypeError: Cannot read property 'enter' of undefined
    at startup.js (startup.js:123)
    at fileEvaluate (modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343)
    at require (modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238)
    at index.js (index.js:1)
    at fileEvaluate (modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343)
    at require (modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238)
    at plugins.js (plugins.js:1)
    at fileEvaluate (modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343)
    at require (modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238)
    at app.js?hash=73b5c87e93b27442bd1e92ebf527872baaae4343:126187
```

hrath2015 commented:
@jshimko  when I reverted latest commit in router.js https://github.com/reactioncommerce/reaction/commit/28a53fb9f793ce7079a9e8e3b246c9acf167d4a3 
it started loading and working fine. May be something to do with logger.

jshimko commented:
Thanks for the update @hrath2015.  Updated.

zenweasel commented:
@jshimko re: `Error: Cannot find module './build/Release/DTraceProviderBindings'`
I had done `reaction reset -y` and did it again for good measure and still seeing that issue.

zenweasel commented:
So now with the better traceback support, it looks like that error when capturing an order is coming from Shippo(? I don't have Shippo enabled)
```
00:03:42.499Z ERROR Reaction: Cannot read property 'userId' of undefined
  TypeError: Cannot read property 'userId' of undefined
      at shippoConfirmShippingMethodForOrder (imports/plugins/included/shipping-shippo/server/methods/shippo.js:510:33)
      at server/api/hooks.js:54:14
```

jshimko commented:
So I guess an important point worth mentioning here is that none of these unhandled Promise rejections are new or caused by Meteor 1.6.  These were always happening (and are happening in the current release).  The only difference before this PR is Node 4 would just silently swallow the errors when a Promise didn't have a `catch` expression.  Node 8 refuses to do that though.  So I guess I'm just saying things are technically not more broken as a result of changes here.  We just know about the issues now.

zenweasel commented:
@jshimko I don't think the error I pointed out is related to promise handling unless I am missing something

zenweasel commented:
Do we have a plan on moving this forward?

zenweasel commented:
The last bug I encountered is now fixed so I am approving. I am no longer seeing the `Dbuild` error either.

jshimko commented:
The shippo error is now fixed.  The function body for `shippo/confirmShippingMethodForOrder` was being called directly in a hook instead of using...
```js
Meteor.call("shippo/confirmShippingMethodForOrder", orderId)
```
So, [right here](https://github.com/reactioncommerce/reaction/blob/meteor-1.6/imports/plugins/included/shipping-shippo/server/methods/shippo.js#L544
) I changed this...
```js
Hooks.Events.add("onOrderPaymentCaptured", methods["shippo/confirmShippingMethodForOrder"]);
```
To this...
```
Hooks.Events.add("onOrderPaymentCaptured", (orderId) => {
  Meteor.call("shippo/confirmShippingMethodForOrder", orderId)
});
```
because the original was calling [this function](https://github.com/reactioncommerce/reaction/blob/meteor-1.6/imports/plugins/included/shipping-shippo/server/methods/shippo.js#L506) directly instead of `Meteor.call`'ing it.

As far as I know, that was the last issue and this branch is now up to date with master.  Let's get this merged!

aaronjudd commented:
I'm still seeing some client console errors: 
```
TypeError: Cannot read property '_currentElement' of null
    at ReactCompositeComponentWrapper._updateRenderedComponent (modules.js?hash=0bc0d31c274ae46fac4934c4cc6ffe2f4db2fc4a:141040)
    at ReactCompositeComponentWrapper._performComponentUpdate (modules.js?hash=0bc0d31c274ae46fac4934c4cc6ffe2f4db2fc4a:141019)
    at ReactCompositeComponentWrapper.updateComponent (modules.js?hash=0bc0d31c274ae46fac4934c4cc6ffe2f4db2fc4a:140940)
    at ReactCompositeComponentWrapper.receiveComponent (modules.js?hash=0bc0d31c274ae46fac4934c4cc6ffe2f4db2fc4a:140842)
    at Object.receiveComponent (modules.js?hash=0bc0d31c274ae46fac4934c4cc6ffe2f4db2fc4a:133487)
    at ReactCompositeComponentWrapper._updateRenderedComponent (modules.js?hash=0bc0d31c274ae46fac4934c4cc6ffe2f4db2fc4a:141049)
    at ReactCompositeComponentWrapper._performComponentUpdate (modules.js?hash=0bc0d31c274ae46fac4934c4cc6ffe2f4db2fc4a:141019)
    at ReactCompositeComponentWrapper.updateComponent (modules.js?hash=0bc0d31c274ae46fac4934c4cc6ffe2f4db2fc4a:140940)
    at ReactCompositeComponentWrapper.receiveComponent (modules.js?hash=0bc0d31c274ae46fac4934c4cc6ffe2f4db2fc4a:140842)```


This is reproduced with a login, then a logout.

spencern commented:
@aaronjudd I'm unable to reproduce the error you're seeing above, which page were you on?

Screencast here: http://recordit.co/zVrU8gffjx

zenweasel commented:
Just did a fresh clone here and not seeing Aaron's error just logging in and logging out.

aaronjudd commented:
I am unable to recreate the component error after pulling/retesting.  Seems ok.  Did find and report #3356, but don't think it's a blocker and probably not related.
